src/proxy/NearCachedMapProxy.ts

```diff
@@ -138,28 +138,28 @@ export class NearCachedMapProxy<K, V> extends MapProxy<K, V> {
                                   ttl: number | Long = -1,
                                   maxIdle?: number | Long): Promise<V> {
         return super.putIfAbsentInternal(keyData, valueData, ttl, maxIdle)
-            .then<V>(this.invalidateCacheEntryAndReturn.bind(this, keyData));
+            .then<V>((val: V) => this.invalidateCacheEntryAndReturn(keyData, val));
```

Comment:
Minor: We could rename the parameter as `oldValue` (instead of `val`) to make it more readable. Similar thing applies to other places.

BTW, do we need both `.then<V>` and `(val: V)` type annotations? (here and in other places)